src/stactools/sentinel1/grd/metadata_links.py

```diff
@@ -0,0 +1,148 @@
+import os
+from typing import List, Optional
+
+import pystac
+
+from stactools.core.io.xml import XmlElement
+
+from .constants import SAFE_MANIFEST_ASSET_KEY
+
+
+class ManifestError(Exception):
+    pass
+
+
+class MetadataLinks:
+    def __init__(
+        self,
+        granule_href: str,
+    ):
+        self.granule_href = granule_href
+        self.href = os.path.join(granule_href, "manifest.safe")
+
+        root = XmlElement.from_file(self.href)
+        data_object_section = root.find("dataObjectSection")
+        if data_object_section is None:
+            raise ManifestError(
+                f"Manifest at {self.href} does not have a dataObjectSection")
```

Comment:
@maximlamare just wanted to point out https://github.com/bopen/xarray-sentinel/blob/main/xarray_sentinel/esa_safe.py , which seems to be a nice approach for parsing SAFE files using the accompanying .xsd files for XML validation and parsing

Reply:
Once we have contributed all sensors we had in mind, I could take a look and propose an update to account for this.